apps/editor.planx.uk/src/@planx/components/shared/useFormikWithRef.ts

```diff
@@ -0,0 +1,17 @@
+import { FormikProps } from "formik";
+import { MutableRefObject, useEffect } from "react";
+
+/**
+ * Hook to sync a Formik instance with a parent ref.
+ * This enables the FormModal to track the changed/unchanged state for unsaved changes warnings.
+ */
+export function useFormikWithRef<Values>(
+  formik: FormikProps<Values>,
+  formikRef?: MutableRefObject<FormikProps<Values> | null>,
+): void {
+  useEffect(() => {
+    if (formikRef) {
+      formikRef.current = formik;
+    }
+  }, [formik, formikRef]);
+}
```

Comment:
```suggestion
export function useFormikWithRef<T extends FormikValues>(
  formikConfig: FormikConfig<T>,
  formikRef?: MutableRefObject<FormikProps<T> | null>,
) {
  const formik = useFormik<T>({
    validateOnChange: false,
    validateOnBlur: false,
    ...formikConfig,
  });

  useEffect(() => {
    if (formikRef) {
      formikRef.current = formik;
    }
  });

  return formik;
}
```

I think something like this, which acts as a wrapper for the `useFormik()` hook will be a bit more comprehensive. We can now also store shared config here like `validateOnChange` and `validateOnBlur`.